object_tracker/object_tracker.go

```diff
@@ -166,10 +184,44 @@ func (t *myTracker) run(stream gostream.VideoStream, cancelableCtx context.Conte
 			HA, _ := hg.NewHungarianAlgorithm(matchMtx)
 			matches := HA.Execute()
 			// Rename from temporal matches. New det copies old det's label
-			renamedNew := t.RenameFromMatches(matches, namedOld, filteredNew)
+			curDets, newDets := t.RenameFromMatches(matches, namedOld, filteredNew)
+			if len(newDets) > 0 {
+				if t.triggerCancelFunc != nil {
+					t.triggerCancelFunc()
+				}
+				triggerContext, triggerCancelFunc := context.WithCancel(t.cancelContext)
+				t.triggerContext = triggerContext
+				t.triggerCancelFunc = triggerCancelFunc
+
+				t.newInstance.Store(true)
+				t.activeBackgroundWorkers.Add(1)
+
+				go func() {
+					coolDownTimer := time.After(time.Duration(t.coolDown * float64(time.Second)))
+					for {
+						select {
+						case <-coolDownTimer:
+							t.newInstance.Store(false)
+							t.activeBackgroundWorkers.Done()
+							return
+						case <-t.triggerContext.Done():
+							t.activeBackgroundWorkers.Done()
+							return
+						}
+					}
```

Comment:
I don't think you need this for{} loop. A select statement will freeze execution until one of the cases is true. So in your case here, the for loop will not even go through one loop, and will return only once of these two cases is true

Reply:
yes sorry